File: Frontend/Marketplace/model/News.js
/**
 * Модель новости/промо-баннера
 */
Ext.define('Marketplace.model.News', {
    extend: 'Ext.data.Model',

    fields: [
        { name: 'id', type: 'int' },
        { name: 'title', type: 'string' },
        { name: 'content', type: 'string' },
        { name: 'image', type: 'string' },
        { name: 'publishedAt', type: 'date' },
        { name: 'isActive', type: 'boolean' },
        { name: 'author', type: 'string' }
    ]
});

File: Backend/Models/News.cs
using System.ComponentModel.DataAnnotations;

namespace net_chuvgu.Backend.Models;

public class News
{
    public int Id { get; set; }
        
    [Required]
    [StringLength(200)]
    public string Title { get; set; }
        
    public string Content { get; set; }
    public string Image { get; set; }
    public DateTime PublishedAt { get; set; }
    public bool IsActive { get; set; }
    public string Author { get; set; }
}